### ISSUE TITLE
Safeharbor release checklist

### ISSUE THREAD

**Issue by alerque** [lunarmodules/luasocket#360]
I've been trying to get my head around what should be required for the next release. This is to collect my thoughts and anybody else's that chimes in rather than just winging it.

Personally I consider a release urgent (give Lua issues with 5.4.3+), even without making major fixes. In fact I specifically think it should happen BEFORE lots of changes start happening. In spite of never having an official 3.0 release, the current Git HEAD as of before migration to lunarmodules was pretty much the defacto build to use. Many distros had even packaged it and many projects were using the dev release instead of the RC1 tag because the latter was just so obsolete.

As such I believe we need a "safe harbor" style release tag before we change to much around so that anybody whose projects rely on exactly the idiosyncrasies (bugs or otherwise) of the current affairs have something to turn to that gets them up and running with the least disruption to the status quo as possible.

After that we can iterate on bug fixes and features and documentation and project refactoring and whatever else whe feel lite.

Things we think must happen before the next release:

- [x] Lua 5.4.3+ fixes (#331, #334)
- [x] Lua Rocks org migration to new namespace
- [x] CI testing on Unix (Linux & macOS operational) (#356)
- [x] Upload rockspecs automatically (#356)
- [ ] Some sort of rough changelog describing how we got from 3.0-rc1 to 3.0 (#368)
- [x] Republish existing docs directory via GH pages from this namespace and update links

Things we think would be nice to have but won't hold up the tag for if they aren't shaping up in time:

- [ ] CI testing on Windows platform(s)
- [ ] Up to date documentation of everything changed since the latest 2.x release

Others?

**Comment by Tieske** [comment]
switch the docs from the current site to Lunarmodules ghpages branch or /docs folder (and updating links)

**Comment by alerque** [comment]
Done, see #361. I already update the repo settings so as soon as that goes through it should get published.
